src/useHumanbound.ts

```diff
@@ -0,0 +1,116 @@
+import { BigNumber, providers } from "ethers";
+import { useCallback, useEffect, useState } from "react";
+
+import { AlchemyNetworkNames, Web3ChainReference, humanboundDeployments } from "./deployments";
+import { IHumanboundToken, IHumanboundToken__factory } from "./types";
+
+type PartialRecord<K extends keyof any, T> = {
+  [P in K]?: T;
+};
+type AlchemyAPIKeyConfig = PartialRecord<
+  | Web3ChainReference.EIP155_ETHEREUM_MAINNET
+  | Web3ChainReference.EIP155_ETHEREUM_RINKEBY
+  | Web3ChainReference.EIP155_ETHEREUM_KOVAN
+  | Web3ChainReference.EIP155_OPTIMISM
+  | Web3ChainReference.EIP155_OPTIMISM_GOERLI
+  | Web3ChainReference.EIP155_ARBITRUM_ONE
+  | Web3ChainReference.EIP155_ARBITRUM_GOERLI
+  | Web3ChainReference.EIP155_POLYGON_MAINNET
+  | Web3ChainReference.EIP155_POLYGON_MUMBAI,
+  string
+>;
+
+/**
+ * useHumanbound react hook for humanbound token data
+ * Provides access to three functions:
+ *  - hasHBT: @param `address`
+ *  - getHBTIdOfOwner: @param `address`
+ *  - getOwnerOf: @param `tokenId`
```

Comment:
Ouuh actually this is checking one network at a time for `hasHBT` 🤔

Reply:
Yeh it's checking one network at a time because we're assuming the same kind of stack of web3 providers, which are always one network at a time. It's the simplest IMO.